test_conformance/basic/test_if.cpp

```diff
@@ -21,146 +21,120 @@
 #include <sys/types.h>
 #include <sys/stat.h>
 
+#include <algorithm>
+#include <vector>
 
 #include "procs.h"
 
-const char *conditional_kernel_code =
-"__kernel void test_if(__global int *src, __global int *dst)\n"
-"{\n"
-"    int  tid = get_global_id(0);\n"
-"\n"
-"    if (src[tid] == 0)\n"
-"        dst[tid] = 0x12345678;\n"
-"    else if (src[tid] == 1)\n"
-"        dst[tid] = 0x23456781;\n"
-"    else if (src[tid] == 2)\n"
-"        dst[tid] = 0x34567812;\n"
-"    else if (src[tid] == 3)\n"
-"        dst[tid] = 0x45678123;\n"
-"    else if (src[tid] == 4)\n"
-"        dst[tid] = 0x56781234;\n"
-"    else if (src[tid] == 5)\n"
-"        dst[tid] = 0x67812345;\n"
-"    else if (src[tid] == 6)\n"
-"        dst[tid] = 0x78123456;\n"
-"    else if (src[tid] == 7)\n"
-"        dst[tid] = 0x81234567;\n"
-"    else\n"
-"        dst[tid] = 0x7FFFFFFF;\n"
-"\n"
-"}\n";
+namespace {
+const char *conditional_kernel_code = R"(
+__kernel void test_if(__global int *src, __global int *dst)
+{
+    int  tid = get_global_id(0);
+
+    if (src[tid] == 0)
+        dst[tid] = 0x12345678;
+    else if (src[tid] == 1)
+        dst[tid] = 0x23456781;
+    else if (src[tid] == 2)
+        dst[tid] = 0x34567812;
+    else if (src[tid] == 3)
+        dst[tid] = 0x45678123;
+    else if (src[tid] == 4)
+        dst[tid] = 0x56781234;
+    else if (src[tid] == 5)
+        dst[tid] = 0x67812345;
+    else if (src[tid] == 6)
+        dst[tid] = 0x78123456;
+    else if (src[tid] == 7)
+        dst[tid] = 0x81234567;
+    else
+        dst[tid] = 0x7FFFFFFF;
+}
+)";
 
 const int results[] = {
```

Comment:
Could we move this `results` variable from being global to being defined inside the `verify_if` function?